### ISSUE TITLE
Question: RPN Loss vs. Class Loss

### ISSUE THREAD

**Issue by sjlee-me** [ppriyank/Pytorch-CustomDataset-FasterRCNN#1]
First of all, thank you for uploading this great and easy to understand repository. I'm implementing Faster RCNN my own codes ( https://github.com/the-realest-one/faster-rcnn-pytorch ), with help from your repository.

My question is about loss functions.
I read faster-rcnn paper, and I found only one loss function which is a multi-task loss ( https://arxiv.org/pdf/1506.01497.pdf, formula (1))
But when I saw your loss.py, there are 4 loss functions. And I think `rpn_loss_regr` and `rpn_loss_cls_fixed_num` are components for  paper's formula(1).

So my questions are:
1. Are really `rpn_loss_regr` and `rpn_loss_cls_fixed_num` component functions of paper's formula(1) ?
2. Are `class_loss_cls` and `class_loss_regr` in your loss.py component functions of Fast RCNN ? ( I was confused because there is only one loss function definition in the paper, but in your repository, there are loss functions for RPN and for Class. So I guess that the paper just didn't write out loss function of Fast RCNN part, and that `class_loss_cls` and `class_loss_regr` are for Fast RCNN)
3. What is the license of your repository? (MIT, Apache, or something)

Because this is the first time I implement object detection network, my question would be awkward. But if you answer my question, it will be much helpful for me.

Thank you

**Comment by ppriyank** [comment]
Hello

Thanks for reaching out. It's been a while since I last worked on this, so not quite remember, how one-to-one is this implementation with the paper. I have mostly re-implemented the official implementation, so most likely most of the things will be according to the paper. 
1 & 2: I don't exactly remember, hence keeping the issue open, if someone else in the future replies to this, or I will if I again work on this (not anytime soon though) 

3. There is no license. Use it as you want.

**Comment by sjlee-me** [comment]
Thank you for replying.

As you say, I will leave this issue open for a while, hoping someone to reach here. And also if I find solution, I will write here what I find.

Thank you
